### PR TITLE
changes to store track data

### PR DESCRIPTION
**Pull request by alvak1**
Testing instructions

- Run the below command locally which will take a while to complete as it will delete all records from table `service_journey_pattern_links` to add another column.

  `npm run deploy -- --stage ${local_stage}`

- Run the batch processes to load all data from tnds and bods

### CODE REVIEW

File: packages/core/db.ts [hunk 1]
```diff
@@ -164,6 +164,19 @@ export interface ServiceJourneyPatternLinksTable {
     orderInSequence: number;
     fromSequenceNumber: string | null;
     toSequenceNumber: string | null;
+    routeLinkRef: string | null;
+}
+
+export interface RoutesTable {
+    id: Generated<number>;
```
Comment (alainajaffer) [inline packages/core/db.ts:172]: could we have a test written to see if it is getting the data from the xml for tracks?

File: packages/ref-data-uploaders/txc-uploader/txc_processor.py [hunk 2]
```diff
@@ -222,8 +225,35 @@ def iterate_through_journey_patterns_and_run_insert_queries(
...
 
-def insert_admin_area_codes(cursor: aurora_data_api.AuroraDataAPICursor, area_codes, service_id):
-    codes_dict = {f"k{k}":v for k,v in enumerate(area_codes)}
+def iterate_through_routes_and_run_insert_queries(
+    cursor, data: dict, operator_service_id: str, logger
+):
+    route_sections = make_list(data["TransXChange"]["RouteSections"]["RouteSection"])
+
+    routes = []
+    for route_section in route_sections:
```
Comment (alainajaffer) [inline packages/ref-data-uploaders/txc-uploader/txc_processor.py]: could we put this in another function to collect_track_data then return routes

File: packages/core/db.ts (hunk 1 above)
Comment (alvak1) [reply packages/core/db.ts:172]: done

File: packages/ref-data-uploaders/txc-uploader/txc_processor.py (hunk 2 above)
Comment (alvak1) [reply packages/ref-data-uploaders/txc-uploader/txc_processor.py]: done

File: packages/ref-data-uploaders/txc-uploader/txc_processor.py [hunk 3]
```diff
@@ -246,8 +249,47 @@ def iterate_through_journey_patterns_and_run_insert_queries(
...
+        route_links = make_list(route_section["RouteLink"])
+        for route_link in route_links:
+            trackData = route_link.get("Track", None)
+            if trackData is not None:
+                tracks = trackData if isinstance(trackData, list) else [trackData]
+                for track in tracks:
+                    locations = make_list(track["Mapping"]["Location"])
+                    for location in locations:
+                        longitude = location["Longitude"] if location.get("Translation", None) is None else location["Translation"]["Longitude"]            
+                        latitude = location["Longitude"] if location.get("Translation", None) is None else location["Translation"]["Longitude"]
```
Comment (ljones92) [inline packages/ref-data-uploaders/txc-uploader/txc_processor.py]: Should this be "Latitude" rather than "Longitude"?

Comment (alvak1) [reply packages/ref-data-uploaders/txc-uploader/txc_processor.py]: done

**Pull request closed without merging**